Assets/Scripts/PlayerController.cs

```diff
@@ -106,35 +107,58 @@ bool MoveControlByTranslate()
         }
 
         string animationName = "";
-
+        Vector3 nextPoint;
+        float xMod = 0;
+        float yMod = 0;
 		if (Input.GetKey(KeyCode.W)|Input.GetKey(KeyCode.UpArrow)) //前
 		{
+            nextPoint = Vector3.up * TranslateAmount;
+            if (!isNextSpaceBlocked(nextPoint)) {
+                yMod += nextPoint.y;
+            }
             animationName = "WalkBack";
-            this.transform.Translate(Vector3.up*TranslateAmount); 
             clicked = true;
 		}
 		if (Input.GetKey(KeyCode.S) | Input.GetKey(KeyCode.DownArrow)) //后
 		{
+            nextPoint = Vector3.up * -TranslateAmount;
+            if (!isNextSpaceBlocked(nextPoint)) {
+                yMod += nextPoint.y;
+            }
             animationName = "WalkFront";
-            this.transform.Translate(Vector3.up *- TranslateAmount); 
             clicked = true;
 		}
 		if (Input.GetKey(KeyCode.A) | Input.GetKey(KeyCode.LeftArrow)) //左
 		{
+            nextPoint = Vector3.right * -TranslateAmount;
+            if (!isNextSpaceBlocked(nextPoint)) {
+                xMod += nextPoint.x;
+            }
             animationName = "WalkLeft";
-            this.transform.Translate(Vector3.right *-TranslateAmount); 
             clicked = true;
 		}
 		if (Input.GetKey(KeyCode.D) | Input.GetKey(KeyCode.RightArrow)) //右
 		{
+            nextPoint = Vector3.right * TranslateAmount;
+            if (!isNextSpaceBlocked(nextPoint)) {
+                xMod += nextPoint.x;
+            }
             animationName = "WalkRight";
-            this.transform.Translate(Vector3.right * TranslateAmount);
             clicked = true;
         }
-
+        // Use the modifiers to generate a velocity for the RigidBody.
+        rb.velocity = new Vector2(xMod, yMod) * 100f;
         setAnimation(animationName);
 
         if (clicked) return true;
         return false;
     }
+
+    RaycastHit2D isNextSpaceBlocked(Vector3 movementVector) {
```

Comment:
see if this can be a boolean function instead, unless you want to actually use the resulting hit point from the raycast (in which case a name change might be in order)

Reply:
changed